<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EditVarDialog</class>
 <widget class="QDialog" name="EditVarDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>327</width>
    <height>392</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Add variable</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../../sharedcode/resources.qrc">
    <normaloff>:/resources/RES/ffficons/add.png</normaloff>:/resources/RES/ffficons/add.png</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <property name="bottomMargin">
    <number>9</number>
   </property>
   <item>
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="0" column="0">
      <widget class="QLabel" name="varNameLabel">
       <property name="text">
        <string>Variable name:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="varTypeLabel">
       <property name="text">
        <string>Variable type:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="varName"/>
     </item>
     <item row="1" column="1">
      <widget class="QComboBox" name="varType">
       <item>
        <property name="text">
         <string>Integer</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Real</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>String</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Boolean</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>2D point</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>3D point</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Color</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Function</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Image</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Class</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Vector path</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>...</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="PropertyTypeGB">
     <property name="title">
      <string>Variable kind</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QRadioButton" name="radioButton">
        <property name="text">
         <string>Single variable</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="radioButton_2">
        <property name="text">
         <string>Array of type</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="menuWithChoices">
        <property name="text">
         <string>Menu with fixed choices</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="InitValLabel">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="font">
      <font>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Initital value(s):</string>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_9">
     <property name="spacing">
      <number>0</number>
     </property>
     <item>
      <widget class="QStackedWidget" name="PageWidget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="currentIndex">
        <number>6</number>
       </property>
       <widget class="QWidget" name="IntegerValue">
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <property name="spacing">
          <number>3</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="label_10">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Integer:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="IntegerSpinBox">
           <property name="minimum">
            <number>-999999999</number>
           </property>
           <property name="maximum">
            <number>999999999</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="RealValue">
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <property name="spacing">
          <number>3</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="label_11">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Real:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="realSpinBox">
           <property name="decimals">
            <number>6</number>
           </property>
           <property name="minimum">
            <double>-999999999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>999999999.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="TextBoxPage">
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <property name="spacing">
          <number>3</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="label_12">
           <property name="text">
            <string>String:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="StringBox"/>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="BooleanValue">
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <property name="spacing">
          <number>3</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QCheckBox" name="BoolBox">
           <property name="text">
            <string>Check for true, uncheck for false</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="PointBoxPage">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="spacing">
          <number>3</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="label_4">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>X:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="p2x">
           <property name="minimum">
            <double>-999999999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>999999999.000000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_5">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Y:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="p2y"/>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="point3D">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <property name="spacing">
          <number>3</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="label_6">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>X:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="p3x">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimum">
            <double>-999999999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>999999999.000000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_7">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Y:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="p3y">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimum">
            <double>-999999999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>999999999.000000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_8">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Z:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="p3z">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimum">
            <double>-999999999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>999999999.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="Color">
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <property name="spacing">
          <number>3</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QToolButton" name="SetColorToolButton">
           <property name="text">
            <string>Select</string>
           </property>
           <property name="icon">
            <iconset resource="../../sharedcode/resources.qrc">
             <normaloff>:/resources/RES/ffficons/palette.png</normaloff>:/resources/RES/ffficons/palette.png</iconset>
           </property>
           <property name="toolButtonStyle">
            <enum>Qt::ToolButtonTextBesideIcon</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QFrame" name="ColorFrame">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="palette">
            <palette>
             <active>
              <colorrole role="Base">
               <brush brushstyle="SolidPattern">
                <color alpha="255">
                 <red>255</red>
                 <green>255</green>
                 <blue>255</blue>
                </color>
               </brush>
              </colorrole>
              <colorrole role="Window">
               <brush brushstyle="SolidPattern">
                <color alpha="255">
                 <red>255</red>
                 <green>255</green>
                 <blue>255</blue>
                </color>
               </brush>
              </colorrole>
             </active>
             <inactive>
              <colorrole role="Base">
               <brush brushstyle="SolidPattern">
                <color alpha="255">
                 <red>255</red>
                 <green>255</green>
                 <blue>255</blue>
                </color>
               </brush>
              </colorrole>
              <colorrole role="Window">
               <brush brushstyle="SolidPattern">
                <color alpha="255">
                 <red>255</red>
                 <green>255</green>
                 <blue>255</blue>
                </color>
               </brush>
              </colorrole>
             </inactive>
             <disabled>
              <colorrole role="Base">
               <brush brushstyle="SolidPattern">
                <color alpha="255">
                 <red>255</red>
                 <green>255</green>
                 <blue>255</blue>
                </color>
               </brush>
              </colorrole>
              <colorrole role="Window">
               <brush brushstyle="SolidPattern">
                <color alpha="255">
                 <red>255</red>
                 <green>255</green>
                 <blue>255</blue>
                </color>
               </brush>
              </colorrole>
             </disabled>
            </palette>
           </property>
           <property name="autoFillBackground">
            <bool>true</bool>
           </property>
           <property name="frameShape">
            <enum>QFrame::Panel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Sunken</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="FunctionPage">
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <property name="spacing">
          <number>3</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="FunctionLabel">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Empty function</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="WriteFunctionButton">
           <property name="text">
            <string>Write</string>
           </property>
           <property name="icon">
            <iconset resource="../../sharedcode/resources.qrc">
             <normaloff>:/resources/RES/ffficons/page_edit.png</normaloff>:/resources/RES/ffficons/page_edit.png</iconset>
           </property>
           <property name="toolButtonStyle">
            <enum>Qt::ToolButtonTextBesideIcon</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="SelectFunctionButton">
           <property name="text">
            <string>Select</string>
           </property>
           <property name="icon">
            <iconset resource="../../sharedcode/resources.qrc">
             <normaloff>:/resources/RES/ffficons/folder.png</normaloff>:/resources/RES/ffficons/folder.png</iconset>
           </property>
           <property name="toolButtonStyle">
            <enum>Qt::ToolButtonTextBesideIcon</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="Resource">
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <property name="spacing">
          <number>3</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="ResName_label">
           <property name="text">
            <string>(Not selected)</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="NullresourceButton">
           <property name="text">
            <string>NULL</string>
           </property>
           <property name="icon">
            <iconset resource="../../sharedcode/resources.qrc">
             <normaloff>:/resources/RES/ffficons/cross.png</normaloff>:/resources/RES/ffficons/cross.png</iconset>
           </property>
           <property name="toolButtonStyle">
            <enum>Qt::ToolButtonTextBesideIcon</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="SelectResourceButton">
           <property name="text">
            <string>Select</string>
           </property>
           <property name="icon">
            <iconset resource="../../sharedcode/resources.qrc">
             <normaloff>:/resources/RES/ffficons/folder.png</normaloff>:/resources/RES/ffficons/folder.png</iconset>
           </property>
           <property name="toolButtonStyle">
            <enum>Qt::ToolButtonTextBesideIcon</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="AddToArrayButton">
       <property name="text">
        <string>Add</string>
       </property>
       <property name="icon">
        <iconset resource="../../sharedcode/resources.qrc">
         <normaloff>:/resources/RES/ffficons/add.png</normaloff>:/resources/RES/ffficons/add.png</iconset>
       </property>
       <property name="shortcut">
        <string>Return</string>
       </property>
       <property name="toolButtonStyle">
        <enum>Qt::ToolButtonIconOnly</enum>
       </property>
       <property name="autoRaise">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="DelFromArray">
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="../../sharedcode/resources.qrc">
         <normaloff>:/resources/RES/ffficons/delete.png</normaloff>:/resources/RES/ffficons/delete.png</iconset>
       </property>
       <property name="shortcut">
        <string>Del</string>
       </property>
       <property name="autoRaise">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QListWidget" name="ArrayList">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="editTriggers">
      <set>QAbstractItemView::NoEditTriggers</set>
     </property>
     <property name="dragEnabled">
      <bool>true</bool>
     </property>
     <property name="dragDropMode">
      <enum>QAbstractItemView::DragDrop</enum>
     </property>
     <property name="defaultDropAction">
      <enum>Qt::MoveAction</enum>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::ExtendedSelection</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_10">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="OKbutton">
       <property name="text">
        <string>OK</string>
       </property>
       <property name="default">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="CancelButton">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../../sharedcode/resources.qrc"/>
  <include location="../../sharedcode/resources.qrc"/>
  <include location="../../sharedcode/resources.qrc"/>
  <include location="../../sharedcode/resources.qrc"/>
  <include location="../../sharedcode/resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>CancelButton</sender>
   <signal>clicked()</signal>
   <receiver>EditVarDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>295</x>
     <y>338</y>
    </hint>
    <hint type="destinationlabel">
     <x>474</x>
     <y>243</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
